swap_meet/vendor.py

```diff
@@ -1,2 +1,88 @@
+from typing import ItemsView
+
 class Vendor:
-    pass
+
+    def __init__(self, inventory = []):
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self, category):
+        items_in_category = []
+
+        for item in self.inventory:
+            if item.category == category:
+                items_in_category.append(item)
+
+        return items_in_category
+
+    def swap_items(self, other, my_item, their_item):
+
+        if not my_item in self.inventory or not their_item in other.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+        self.inventory.append(their_item)
+        other.inventory.remove(their_item)
+        other.inventory.append(my_item)
+        return True
+
+    
+    def swap_first_item(self, other):
+        if len(self.inventory) == 0 or len(other.inventory) == 0:
```

Comment:
A more "pythonic" way to express checking for empty lists is
```python
        if not self.inventory or not other.inventory:
```
which takes advantage of the falsiness of empty lists.